packages/protocol-kit/src/contracts/SafeWebAuthnSignerFactory/v1.4.1/SafeWebAuthnSignerFactoryContract_v1_4_1.ts

```diff
@@ -0,0 +1,73 @@
+import SafeWebAuthnSignerFactoryBaseContract from '@safe-global/protocol-kit/contracts/SafeWebAuthnSignerFactory/SafeWebAuthnSignerFactoryBaseContract'
+import {
+  SafeVersion,
+  SafeWebAuthnSignerFactoryContract_v1_4_1_Abi,
+  SafeWebAuthnSignerFactoryContract_v1_4_1_Contract,
+  SafeWebAuthnSignerFactoryContract_v1_4_1_Function,
+  SafeWebAuthnSignerFactory_1_4_1_ContractArtifacts
+} from '@safe-global/safe-core-sdk-types'
+import SafeProvider from '@safe-global/protocol-kit/SafeProvider'
+
+/**
+ * SafeWebAuthnSignerFactoryContract_v1_4_1  is the implementation specific to the Safe Proxy Factory contract version 1.4.1.
+ *
+ * This class specializes in handling interactions with the Safe Proxy Factory contract version 1.4.1 using Ethers.js v6.
+ *
+ * @extends SafeWebAuthnSignerFactoryBaseContract<SafeWebAuthnSignerFactoryContract_v1_4_1_Abi> - Inherits from SafeWebAuthnSignerFactoryBaseContract with ABI specific to Safe Proxy Factory contract version 1.4.1.
+ * @implements SafeWebAuthnSignerFactoryContract_v1_4_1_Contract - Implements the interface specific to Safe Proxy Factory contract version 1.4.1.
+ */
+class SafeWebAuthnSignerFactoryContract_v1_4_1
+  extends SafeWebAuthnSignerFactoryBaseContract<SafeWebAuthnSignerFactoryContract_v1_4_1_Abi>
+  implements SafeWebAuthnSignerFactoryContract_v1_4_1_Contract
+{
+  safeVersion: SafeVersion
+
+  /**
+   * Constructs an instance of SafeWebAuthnSignerFactoryContract_v1_4_1
+   *
+   * @param chainId - The chain ID where the contract resides.
+   * @param safeProvider - An instance of SafeProvider.
+   * @param customContractAddress - Optional custom address for the contract. If not provided, the address is derived from the Safe deployments based on the chainId and safeVersion.
+   * @param customContractAbi - Optional custom ABI for the contract. If not provided, the default ABI for version 1.4.1 is used.
+   */
+  constructor(
+    chainId: bigint,
+    safeProvider: SafeProvider,
+    customContractAddress?: string,
+    customContractAbi?: SafeWebAuthnSignerFactoryContract_v1_4_1_Abi
+  ) {
+    const safeVersion = '1.4.1'
+    const defaultAbi = SafeWebAuthnSignerFactory_1_4_1_ContractArtifacts.abi
+
+    super(chainId, safeProvider, defaultAbi, safeVersion, customContractAddress, customContractAbi)
+
+    this.safeVersion = safeVersion
```

Comment:
Any particular reason why its not just `this.safeVersion = '1.4.1'`?

Reply:
It is because `super(...)` call must be the first statement in the constructor. after `super(...)` call we can set the `this.safeVersion` value